cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowToken.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+import java.util.Map;
+
+/**
+ * Interface to represent the data that is transferred from one node to the next node in the {@link Workflow}
+ */
+public interface WorkflowToken {
+  /**
+   *
+   * @return the Hadoop MapReduce counters from the previous MapReduce program in the Workflow
+   */
+  public Map<String, Map<String, Long>> getMapReduceCounters();
```

Comment:
I think this should be @Nullable if the counters are not set in the token. And the javadoc should mention what it returns (null or empty map, I vote for null).